### PR TITLE
T4156: Add bootfile-size option

### PR DESCRIPTION
**Pull request by tacerus**
Signed-off-by: Georg <georg@lysergic.dev>

## Change Summary
This adds a `bootfile-size` setting allowing to easily set DHCP option 13.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4156

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly -> shall I do that after the PR is accepted or attach a PR for the documentation to this one?

### CODE REVIEW

File: interface-definitions/dhcp-server.xml.in [hunk 1]
```diff
@@ -142,6 +142,11 @@
                 boot file is to be loaded</help>
                     </properties>
                   </leafNode>
+		  <leafNode name="bootfile-size">
+		    <properties>
+		      <help>Bootstrape file size in 512 byte blocks</help>
```
Comment (dmbaturin) [inline interface-definitions/dhcp-server.xml.in]: Typo! Also, I have to admit that I haven't used that option, I only used the boot-file one and TFTP clients picked it up just fine. I wonder why it's in 512 byte blocks, and can the CLI allow the user to specify it in k/mbytes instead and convert it to blocks internally?

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: interface-definitions/dhcp-server.xml.in (hunk 1 above)
Comment (tacerus) [reply interface-definitions/dhcp-server.xml.in]: Sorry about that - repaired the typo. Whilst I assume most users using this relatively obscure option have the right number handy already, a conversion feature would indeed be handy - but I unfortunately don't know how to implement that.

Comment (dmbaturin) [reply interface-definitions/dhcp-server.xml.in]: I think it's best to add a fixup in the script that is calling that template: https://github.com/vyos/vyos-1x/blob/current/src/conf_mode/dhcp_server.py#L281

The `generate()` function takes the config dict as an argument, so you can check for that option there and modify the value.

**Comment by sarthurdev** [comment]
Can the indentation be updated to use spaces?

**Comment by tacerus** [comment]
Hi,

as already stated in IRC, I am unsure a mathematical conversion is feasible.

I learnt to use `du -B512 <file>` to acquire the number to fill the DHCP option with. The `du -B512 <file>` operation reports the block size based on the actual file on disk.
However dividing by 512, which is how I thought of doing a conversion based purely on a number, not on a file, as intended here, would return a result similar to `du --apparent-size -B512 <file>`, which can be way off the regular `du -B512 <file>` "calculation".
It is possible that a purely mathematical conversion based on a number is ok, however I could not find anything confirming or denying this - I can only tell from my experience that a number taken from `du -B512 <file>` would work with PXE clients.

Do we still want to implement the conversion feature?
Does someone have a reference as to whether a mathematically converted number is "good enough" in DHCP option 13?

File: interface-definitions/dhcp-server.xml.in (hunk 1 above)
Comment (tacerus) [reply interface-definitions/dhcp-server.xml.in]: Replied in main thread (https://github.com/vyos/vyos-1x/pull/1148#issuecomment-1027297794).

**Comment by tacerus** [comment]
> Can the indentation be updated to use spaces?

OK.

**Comment by dmbaturin** [comment]
@tacerus I think it's good, could you resolve the merge conflicts?

**Comment by tacerus** [comment]
Done!

**Comment by tacerus** [comment]
Thanks, @dmbaturin. Shall I follow up with a PR to [vyos-documentation](https://github.com/vyos/vyos-documentation) or is there some other process for it (Phabricator)?

**Comment by dmbaturin** [comment]
Yes, please!

**Pull request merged**